datadog_checks_downloader/datadog_checks/downloader/download.py

```diff
@@ -86,52 +79,41 @@ def __init__(
         # NOTE: Build a TUF updater which stores metadata in (1) the given
         # directory, and (2) uses the following mirror configuration,
         # respectively.
-        # https://github.com/theupdateframework/tuf/blob/aa2ab218f22d8682e03c992ea98f88efd155cffd/tuf/client/updater.py#L628-L683
         # NOTE: This updater will store files under:
         # os.path.join(REPOSITORIES_DIR, REPOSITORY_DIR)
         self.__updater = Updater(
-            REPOSITORY_DIR,
-            {
-                'repo': {
-                    'url_prefix': repository_url_prefix,
-                    'metadata_path': 'metadata.staged',
-                    'targets_path': 'targets',
-                    'confined_target_dirs': [''],
-                }
-            },
+            metadata_dir=os.path.join(REPOSITORIES_DIR, REPOSITORY_DIR, 'metadata/current'),
+            metadata_base_url=f'{repository_url_prefix}/metadata.staged/',
+            target_base_url=f'{repository_url_prefix}/targets/',
+            target_dir=self.__targets_dir,
         )
 
+        # Increase requests timeout.
+        # There's no officially supported way to do this without either writing our own
+        # fetcher from scratch or relying on internals. We're choosing the latter for now.
+        # - https://github.com/theupdateframework/python-tuf/blob/v2.0.0/tuf/ngclient/updater.py#L99
+        # - https://github.com/theupdateframework/python-tuf/blob/v2.0.0/tuf/ngclient/_internal/requests_fetcher.py#L49
+        self.__updater._fetcher.socket_timeout = 60
+
         # NOTE: Update to the latest top-level role metadata only ONCE, so that
         # we use the same consistent snapshot to download targets.
         self.__updater.refresh()
 
     def __download_with_tuf(self, target_relpath):
-        target = self.__updater.get_one_valid_targetinfo(target_relpath)
-        updated_targets = self.__updater.updated_targets((target,), self.__targets_dir)
+        target = self.__updater.get_targetinfo(target_relpath)
```

Comment:
You need to handle the [possibility](https://github.com/theupdateframework/python-tuf/blob/45271efe3b4e4895fb6e7a4ed4fb4f234b188cfd/tuf/ngclient/updater.py#L478) that `target` could be `None` (i.e., not found), and ideally also add a test case for it.